builder/dockerfile/evaluator.go

```diff
@@ -192,12 +111,30 @@ type dispatchState struct {
 	imageID    string
 	baseImage  builder.Image
 	stageName  string
+	buildArgs  *buildArgs
 }
 
-func newDispatchState() *dispatchState {
-	return &dispatchState{runConfig: &container.Config{}}
+type stageBuild struct {
+	state   *dispatchState
+	shlex   *ShellLex
+	builder *Builder
+	source  builder.Source
+}
```

Comment:
Could this be renamed back to `dispatchRequest` (and with the variable name in the dispatch functions back to `req`). I think that will help focus the diff on the changes related to the new typed instructions.